lib/Sema/DerivedConformances.h

```diff
@@ -71,6 +71,25 @@ ValueDecl *getDerivableRequirement(TypeChecker &tc,
                                    NominalTypeDecl *nominal,
                                    ValueDecl *requirement);
 
+
+/// Derive a CaseIterable type witness for an enum if it has no associated
+/// values for any of its cases.
+///
+/// \returns the derived member, which will also be added to the type.
+ValueDecl *deriveCaseIterable(TypeChecker &tc,
+                                 Decl *parentDecl,
+                                 NominalTypeDecl *type,
+                                 ValueDecl *assocType);
+
+/// Derive a CaseIterable type witness for an enum if it has no associated
+/// values for any of its cases.
+///
+/// \returns the derived member, which will also be added to the type.
```

Comment:
Are the comments on these two functions supposed to be identical? (I also find it a bit strange that the functions are named identically, but I realize that's a pattern that already exists in this file — nonetheless, did you consider something like `deriveCaseIterable_property_allCases` and `deriveCaseIterable_associatedtype_AllCases`? Is there some motivation I'm missing here for this naming convention?)